### PR TITLE
debounce decorator

### PR DESCRIPTION
**Pull request by snewcomer**
A little more luck with this one.  :smile:  @pzuraq

### CODE REVIEW

File: addon/index.js [hunk 1]
```diff
@@ -28,4 +30,36 @@ function decoratorWithReturnValue(fn) {
  * ```
  * @function
  */
-export const next = decoratorWithReturnValue(run.next);
\ No newline at end of file
+export const next = decoratorWithReturnValue(run.next);
+
+
+function decoratorWithDebounce(fn) {
```
Comment (pzuraq) [inline addon/index.js]: Did you extract this macro for a reason? I think we could probably use it for `throttle`, just wondering

File: tests/unit/runloop-test.js [hunk 2]
```diff
@@ -37,4 +37,36 @@ test('"run.next" works with arguments', function(assert) {
...
+  class Foo {
+    @debounce(50)
+    nextMe() {
+      assert.ok(true);
+      done();
+    }
+  }
+
+  let obj = new Foo();
+  obj.nextMe();
```
Comment (pzuraq) [inline tests/unit/runloop-test.js]: It doesn't look like we're testing the debounce functionality here, can we call these functions at least twice to make sure they're working properly?

**Comment by snewcomer** [comment]
@pzuraq Great feedback! Added the immediate flag and improved the tests.

File: tests/unit/runloop-test.js [hunk 3]
```diff
@@ -37,4 +37,53 @@ test('"run.next" works with arguments', function(assert) {
...
+  obj.nextMe('wat');
+});
+
+test('"run.debounce" works with immediate flag', function(assert) {
+  assert.expect(1);
+
+  class Foo {
+    @debounce(50, true)
+    nextMe(arg) {
+      assert.equal(arg, 'wat', 'immediate decorator works and debounces second function call');
```
Comment (snewcomer) [inline tests/unit/runloop-test.js]: notice no `done()` call here.

**Comment by pzuraq** [comment]
Thanks again!

**Pull request merged**